enterprise_gateway/services/processproxies/conductor.py

```diff
@@ -0,0 +1,374 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+"""Kernel managers that operate against a remote process."""
+
+import os
+import signal
+import json
+import time
+import tornado
+import logging
+import subprocess
+import errno
+import socket
+from tornado import web
+from jupyter_client import launch_kernel, localinterfaces
+from .processproxy import RemoteProcessProxy
+
+pjoin = os.path.join
+
+try:
+    from urlparse import urlparse
+except ImportError:
+    from urllib.parse import urlparse
+
+local_ip = localinterfaces.public_ips()[0]
+poll_interval = float(os.getenv('EG_POLL_INTERVAL', '0.5'))
+max_poll_attempts = int(os.getenv('EG_MAX_POLL_ATTEMPTS', '10'))
+
+
+class ConductorClusterProcessProxy(RemoteProcessProxy):
+    initial_states = {'SUBMITTED', 'WAITING', 'RUNNING'}
+    final_states = {'FINISHED', 'KILLED', 'RECLAIMED'}  # Don't include FAILED state
+
+    def __init__(self, kernel_manager, proxy_config):
+        super(ConductorClusterProcessProxy, self).__init__(kernel_manager, proxy_config)
+        self.application_id = None
+        self.conductor_endpoint = proxy_config.get('conductor_endpoint', kernel_manager.parent.parent.conductor_endpoint)
+
+    def launch_process(self, kernel_cmd, **kw):
+        """ Launches the kernel process.  Prior to invocation, connection files will be distributed to each applicable
+            Conductor node so that its in place when the kernel is started.  This step is skipped if pull or socket modes 
+            are configured, which results in the kernel process determining ports and generating encoding key.
+            Once started, the method will poll the application (after discovering the application ID via the
+            kernel ID) until host is known.  Note that this polling may timeout and result in a 503 Http error (Service 
+            unavailable).
+            Once the host is determined the connection file is retrieved. If pull mode is configured, the remote file is 
+            copied locally and member variables are loaded based on its contents.  If socket mode is configured, the
+            kernel launcher sends the connection information - which is then written out upon its reception.  If push
+            mode is configured, the kernel manager's IP is updated to the selected node.
+        """
```

Comment:
@charlieeeeeee - please remove all but perhaps the first sentence.  This comment is old and primarily incorrect and we need to update the YarnClusterProcessProxy comment as well.  (Feel free to submit that same update if you like!)
If you want, you could add some comment about how the command is generated via the update method.